reefperf/node_connection.py

```diff
@@ -23,17 +25,35 @@ def __init__(self, keys, hostname, username):
         priv_key_type = keys['keys_type']
         priv_key_str = keys['priv_key_str']
         priv_key_stream = StringIO(priv_key_str)
-        priv_key = self.KEY_CLASSES[priv_key_type].from_private_key(priv_key_stream)
+        self.priv_key = self.KEY_CLASSES[priv_key_type].from_private_key(priv_key_stream)
+        self._hostname = hostname
+        self._username = username
+        self._client = None
+
+    @property
+    def client(self):
+        if self._client is not None:
+            return self._client
         self._client = SSHClient()
         self._client.set_missing_host_key_policy(AutoAddPolicy())
-        self._client.connect(hostname, username=username, pkey=priv_key)
+        self._client.connect(self._hostname, username=self._username, pkey=self.priv_key)
+        return self._client
 
     def exec_command(self, command_str):
-        _, stdout, _ = self._client.exec_command(command_str)
+        _, stdout, _ = self.client.exec_command(command_str)
         return stdout
 
     def scp_file(self, path_to_file):
         raise NotImplementedError()
 
+    def dump_private_key_to_file(self, node_name):
+        key_path = f'/tmp/{node_name}'
+        if os.path.isfile(key_path):
+            os.remove(key_path)
+        with open(key_path, 'w') as key_file:
+            self.priv_key.write_private_key(key_file)
+        os.chmod(key_path, 0o400)
```

Comment:
between this line and a line above it is possible to steal the file. create an empty file first, chmod it and then write to it

Reply:
With chmod 400 python cannot write to file. Anyway the phenomenon that you described isn't likely